tesseract-server/src/handlers/metadata.rs

```diff
@@ -198,7 +207,38 @@ pub fn do_members(
 }
 
 
+pub fn apply_filter_term(df: DataFrame, term: &str) -> DataFrame {
+    debug!("Filtering dataframe by: {}", term);
+
+    // Collect all column names with text data
+    let columns: Vec<String> = (&df.columns).iter()
+        .filter_map(|column| match column.column_data {
+            ColumnData::Text(_) => Some(column.name.clone()),
+            ColumnData::NullableText(_) => Some(column.name.clone()),
+            _ => None,
+        })
+        .collect();
+
+    df.drain_filter(|datum, _| {
+        columns.iter().all(|name| {
```

Comment:
It looks like you're saying that you'll remove a row (`row` should probably be part of `drain_filter`'s name) if:
- for every text column
- the text value doesn't contain the `filter pattern`.

As a first feeling, I think I would have expected `any` instead of `all`. If I'm understanding correctly, `all` implies that all text columns would need to contain or not contain the `filter pattern` in order for something to happen to a row. I might be misunderstanding something here though.

Reply:
Also, I guess I would have expected that you would filter on one particular column?